### PR TITLE
Towards an independent javalib #1

### PR DESCRIPTION
**Pull request by ekrich**
See: https://github.com/scala-native/scala-native/issues/2885

The Java Collections in Scala.js have been much improved and to take advantage of these improvement we need update `java.util.ScalaOps` to be more inline with the Scala.js setup.

We more recently updated the Java Map hierarchy to be able to take advantage of the rewritten `java.util.IdentityHashMap`.

This PR is just to help push us into the correct direction. Some guidelines are on the main issue page.

Collections updated:

- [x] Updated `java.util.Collections`
- [x] Updated `java.util.AbstractMap`
- [x] Removed `java.util.NavigableView` not used, package private, and not in Java
- [x] Removed `java.util.ScalaCompatOps` only used in `NavigableView` above
- [x] Updated `java.util.ScalaOps` with the one section that requires no `Predef` commented out
- [x] Revert `javalib` compile setting
- [x] Check `Collections` tests
- [x] Check `AbstractMap` tests
- [x] Move AssertThrows and deprecate ThrowsHelper (this will help some future poor soul) - large number of files

### CODE REVIEW

File: project/Settings.scala [hunk 1]
```diff
@@ -438,6 +438,7 @@ object Settings {
     // We put our classes on scalac's `javabootclasspath` so that it uses them
     // when compiling rather than the definitions from the JDK.
     recompileAllOrNothingSettings,
+    scalacOptions -= "-Xfatal-warnings", // temp for ScalaOps deprecation
```
Comment (armanbilge) [inline project/Settings.scala:441]: How temporary? We should not merge the PR with this.

Comment (ekrich) [reply project/Settings.scala:441]: It is WIP, with a plan to fix the sites and remove all the deprecated functions. Is that ok?

Comment (armanbilge) [reply project/Settings.scala:441]: Sounds like a good plan, feel free to mark the PR as a draft until it's ready for review.

Comment (ekrich) [reply project/Settings.scala:441]: Didn't see the option. Says WIP in the title.

**Comment by ekrich** [comment]
These should be able to be merged to the `0.4.x` branch to make future changes easier.

**Review by WojciechMazur — APPROVED** [review]
That's a piece of hard work! Thank you

**Pull request merged**